File: Python/Searching/BinarySearch.py
def BinarySearch(arr: list, target):
    num_min = 0
    num_max = len(arr)
    while (num_min <= num_max):
        mid = int((num_min + num_max)/2)
        if arr[mid] == target: return arr[mid]
        if target > arr[mid]: num_min = mid
        else: num_max = mid
    return -1

File: Searching/Rust/Search/src/BinarySearch.rs
pub fn BinarySearch(arr: &[i128], target: i128) -> i128{
	let mut num_min: usize = 0;
	let mut num_max: usize = arr.len() as usize;

	let mut ans: i128 = -1;

	while num_min <= num_max{
		let num_mid: usize = (num_min + num_max)/2;
		if arr[num_mid] == target{
			let ans = arr[num_mid];
			break;
		}
		if target > arr[num_mid]{
			num_min = num_mid + 1;
		} else {
			num_max = num_mid - 1;
		}
	}
	ans
}